src/views/Login.vue

```diff
@@ -45,15 +50,36 @@
 </template>
 
 <script>
+import BaseButton from '@/components/BaseButton.vue';
+import authService from '@/api/authService';
+import { get } from 'lodash';
+
 export default {
+  components: { BaseButton },
   name: 'Login',
   data() {
     return {
+    loading: false,
      model: {
-      email:'',
-      password:''
+     email:'',
+     password:''
      },
     }
-  }
+  },
+  methods: {
+    async handleSubmit() {
+ try {
+      this.loading = true;
+        let {data} =  await authService.login(this.model)
+        const token = get(data, 'token', '')
+        authService.setToken(token)
+      } catch (error) {
+        console.log(error)
+      } finally {
+        this.loading = false;
+      }
+      this.$router.push('/');
```

Comment:
I would push the route only if the login is successful (after setting the token). Else, display a error message